File: rxjava-core/src/main/java/rx/concurrency/ExecutorScheduler.java
package rx.concurrency;

import java.util.concurrent.Executor;
import java.util.concurrent.ScheduledExecutorService;

/**
 * Deprecated. Package changed from rx.concurrency to rx.schedulers.
 * 
 * @deprecated Use {@link rx.schedulers.ExecutorScheduler} instead. This will be removed before 1.0 release.
 */
@Deprecated
public class ExecutorScheduler extends rx.schedulers.ExecutorScheduler {

    @Deprecated
    public ExecutorScheduler(Executor executor) {
        super(executor);
    }

    @Deprecated
    public ExecutorScheduler(ScheduledExecutorService executor) {
        super(executor);
    }

}

File: language-adaptors/rxjava-scala/src/main/scala/rx/lang/scala/concurrency/ExecutorScheduler.scala
package rx.lang.scala.concurrency

import rx.lang.scala.Scheduler
import java.util.concurrent.{ScheduledExecutorService, Executor}

object ExecutorScheduler {

  /**
  * Returns a [[rx.lang.scala.Scheduler]] that queues work on an `java.util.concurrent.Executor`.
  *
  * Note that this does not support scheduled actions with a delay.
  */
  def apply(executor: Executor): ExecutorScheduler =  {
    new ExecutorScheduler(rx.concurrency.Schedulers.executor(executor))
  }
}


class ExecutorScheduler private[scala] (val asJavaScheduler: rx.concurrency.ExecutorScheduler)
  extends Scheduler {}



